### PR TITLE
#309 Add options to channels

### PR DESCRIPTION
**Pull request by RkAiman**

### CODE REVIEW

**Comment by RkAiman** [comment]
Capture d’écran de 2020-11-17 17-45-09

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import Menu from 'components/Menus/Menu.js';
+import Languages from 'services/languages/languages.js';
+import { ChannelType } from 'app/models/Channel';
+import AlertManager from 'services/AlertManager/AlertManager.js';
+import UserService from 'services/user/user.js';
+import ModalManager from 'app/services/Modal/ModalManager';
+import ChannelWorkspaceEditor from 'app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor';
+import Notifications from 'services/user/notifications.js';
+
+type Prpos = {
```
Comment (stephanevieira75) [inline twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx:13]: `Props` or `PropsType`

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  channel: ChannelType;
+};
+
+export default (props: Prpos): JSX.Element => {
```
Comment (stephanevieira75) [inline twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx:17]: If you expect to return html, you could use FC type from react:

``` js
import React, { FC } from 'react';

const ChannelMenu: FC<PropsType> = props => {
// some code here
};
export default ChannelMenu;
```

WDYT ?

File: twake/frontend/src/app/components/Menus/Menu.scss [hunk 3]
```diff
@@ -54,6 +54,10 @@
     color: $grey_dark;
   }
 
+  .menu-text-delicate {
+    color: red !important;
+  }
+
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/components/Menus/Menu.scss:60]: We already have a danger class for menu items: ".danger", please use this one :)

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+      },
+    },
+  ];
+
+  //To do: Add the necessery admin rights
+  if (true) {
+    menu.push({
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.channel_removing'),
+      className: 'menu-text-delicate',
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx:120]: We already have a danger class for menu items: "danger", please use this one :)

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.modify_channel_menu'),
+      onClick: () => {
+        editChannel();
+      },
+    },
+    {
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.channel_leaving'),
+      className: 'menu-text-delicate',
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx:108]: We already have a danger class for menu items: "danger", please use this one :) (this comment will be duplicated several time ^^)

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
Small changes, please also do the changes requested by Stéphane

File: twake/frontend/src/app/components/Menus/Menu.scss (hunk 3 above)
Comment (RkAiman) [reply twake/frontend/src/app/components/Menus/Menu.scss]: Done

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx (hunk 4 above)
Comment (RkAiman) [reply twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx]: Done

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx (hunk 5 above)
Comment (RkAiman) [reply twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx]: Done

File: twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx (hunk 1 above)
Comment (RkAiman) [reply twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx]: Done

**Pull request merged**